docs/content/v0.20/architecture.md

```diff
@@ -1,36 +1,479 @@
+  indexed by GVK (Group, Version, Kind) of the watched objects.
+- Using a common work queue and set of workers, where the key is defined as follows:
+  - Key is a struct instead than a string, and contains the following:
+    - GVK of the informer and lister for the object that generated the
+      event
+    - Structured Name of the object 
+    - For delete event: Shallow copy of the object being deleted. This
+      is required for objects that need to be deleted
+      from the managed clusters (WECs)
+- Starting & stopping informers dynamically based on creation or
+  deletion of CRDs (which add/remove APIs on the WDS).
+- One client connected to the WDS space and one (or more in the future)
+  to connect to one or more OCM shards.
+  - The WDS-connected client is used to start the dynamic
+    informers/listers for most API resources in the WDS
+  - The OCM-connected client is used to start informers/listers for OCM
+    ManagedClusters and to copy/update/remove the wrapped objects
+    into/from the OCM mailbox namespaces.
+
+Currently there are two controllers in the KubeStellar controller
+manager: the placement controller and the status controllor.
+
+### Placement Controller
+
+The Placement controller is responsible for watching workload and 
+placement objects, and wrapping and delivering objects to the ITS
+based on placement policies.
+
+The architecture and the event flow for create/update object events is
+illustrated in Figure 3 (some details might be omitted to make the flow easier
+to understand).
+
+<figure>
+  <img src="./placement-controller.png"  alt="Placement Controller">
+  <figcaption align="center">Figure 3 - Placement Controller</figcaption>
+</figure>
+
+At startup, the controller code sets up the dynamic informers, the event
+handler and the work queue as follows:
+
+- lists all API resources
+- Filters out some resources
+- For each resource:
+  - Creates GVK key
+  - Registers Event Handler
+  - Starts Informer
+  - Indexes informer and lister in a map by GVK key
+- Waits for all caches to sync
+- Starts N workers to process work queue
+
+The reflector is started as part of the informer and watches specific
+resources on the WDS API Server; on create/update/delete object events it
+puts a copy of the object into the local cache. The informer invokes the
+event handler. The handler implements the event handling functions
+(AddFunc, UpdateFunc, DeleteFunc)
+
+A typical event flow for a create/update object event will run as
+follows:
+
+1.  Informer invokes the event handler AddFunc or UpdateFunc
+
+2.  The event handler does some filtering (for example, to ignore update
+    events where the object content is not modified) and then creates a
+    key to store a reference to the object in the work queue. The key
+    contains the GVK key used to retrieve a reference to the informer
+    and lister for that kind of object, and a namespace/name key to
```

Comment:
I would use a plus sign here rather than a slash, between "namespace" and "name".

Reply:
done